RESTserver/src/main/java/com/softserve/if072/restservice/service/TokenService.java

```diff
@@ -0,0 +1,133 @@
+    private final String MESSAGE_DIGEST_ALGORITHM;
+    private final String TOKEN_DELIMITER;
+    private final int TOKEN_VALIDITY_TIME;
+
+    private Environment environment;
+    private UserDAO userDAO;
+    private MessageDigest messageDigest;
+    private HexBinaryAdapter hexBinaryAdapter;
+
+    @Autowired
+    public TokenService(Environment environment, UserDAO userDAO) {
+        this.environment = environment;
+        this.userDAO = userDAO;
+        this.hexBinaryAdapter = new HexBinaryAdapter();
+        this.SECURITY_KEY = environment.getProperty("security.tokenEncryptionKey");
+        this.MESSAGE_DIGEST_ALGORITHM = environment.getProperty("security.messageDigestAlgorithm");
+        this.TOKEN_DELIMITER = environment.getProperty("security.tokenDelimiter");
+        this.TOKEN_VALIDITY_TIME = Integer.parseInt(environment.getProperty("security.tokenValidityTimeInSeconds"));
+        try {
+            this.messageDigest = MessageDigest.getInstance(MESSAGE_DIGEST_ALGORITHM);
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Can't find message digest instance for " + MESSAGE_DIGEST_ALGORITHM, e);
+        }
+    }
+
+    public void validate(CustomAuthenticationToken token) {
+        String[] tokenParts = decodeToken(token.getToken());
+        if (tokenParts != null) {
+            token.setUserName(tokenParts[0]);
+
+            try {
+                token.setExpirateinDate(Long.parseLong(tokenParts[1]));
+            } catch (NumberFormatException e) {
+                LOGGER.warn("Can't parse token expiration data. Date format is incorrect");
+                return;
+            }
+
+            if (new DateTime(token.getExpirateinDate()).isBeforeNow()) {
+                LOGGER.info("Token's validity time expired, new token must be generated");
+                return;
+            }
+
+            token.setConfirmationKey(tokenParts[2]);
+            String expectedKey = buildTokenConfirmationKey(token.getUserName(), token.getExpirateinDate());
+            if (!expectedKey.equals(token.getConfirmationKey())) {
+                LOGGER.warn("Received key does not match the expected one: " + token.getConfirmationKey() + " <> " + expectedKey);
+                return;
+            }
+
+            //if there is no errors, set token as valid
+            token.setValid(true);
+        }
+    }
+
+    public User getUserByToken(CustomAuthenticationToken token) {
+        return userDAO.getByUsername(token.getUserName());
+    }
+
+    public String generateTokenFor(String username) {
+        long expirationDate = DateTime.now().plusSeconds(TOKEN_VALIDITY_TIME).getMillis();
+        String tokenString = buildToken(username, expirationDate);
+        return new String(Base64.encodeBase64(tokenString.getBytes()));
+    }
+
+    private String[] decodeToken(String tokenString) {
+        if (tokenString == null || tokenString.equals("") || !Base64.isBase64(tokenString)) {
```

Comment:
StringUtils.isNotBlank() can be used from Apache Commons library
https://commons.apache.org/proper/commons-lang/